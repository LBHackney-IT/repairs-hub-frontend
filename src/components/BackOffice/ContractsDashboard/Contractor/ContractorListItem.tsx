import Link from 'next/link'
import cx from 'classnames'

interface ContractorListItemProps {
  contractorReference: string
  contractorName?: string
}

const ContractorListItem = ({
  contractorReference,
  contractorName,
}: ContractorListItemProps) => {
  return (
    <Link
      href={{
        pathname: `/backoffice/contractors/${contractorReference}`,
      }}
    >
      <li
        style={{
          cursor: 'pointer',
          border: '5px solid #00664F',
          borderRadius: '20px',
          width: '85%',
        }}
        className={cx('govuk-!-margin-top-3', 'operative-work-order-list-item')}
      >
        <div className="contract-details">
          <h3 className="lbh-heading-h3 lbh-!-font-weight-bold govuk-!-margin-bottom-1">
            {`${contractorName}`}
          </h3>
        </div>
        <div className="govuk-!-margin-0">
          <span className="arrow right"></span>
        </div>
      </li>
    </Link>
  )
}
export default ContractorListItem
