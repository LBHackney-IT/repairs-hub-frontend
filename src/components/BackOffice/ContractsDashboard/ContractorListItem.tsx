import Link from 'next/link'
import cx from 'classnames'

import { dateToStr } from '@/root/src/utils/date'
import Contract from '@/root/src/models/contract'

interface ContractorListItemProps {
  contractorReference: Contract['contractorReference']
}

const ContractorListItem = ({
  contractorReference,
}: ContractorListItemProps) => {
  return (
    // <Link
    //   href={`/backoffice/contracts/${contractorReference.contractReference}`}
    // >
    <li
      style={{
        cursor: 'pointer',
        border: '5px solid #00664F',
        borderRadius: '20px',
        width: '85%',
      }}
      className={cx('govuk-!-margin-top-3', 'operative-work-order-list-item')}
    >
      <div className="contract-details">
        <h3 className="lbh-heading-h3 lbh-!-font-weight-bold govuk-!-margin-bottom-1">
          Reference:{' '}
        </h3>
        <p className="govuk-!-margin-0 govuk-!-margin-bottom-1">
          <strong>Contractor:</strong>{' '}
        </p>
        <p className="govuk-!-margin-0 govuk-!-margin-bottom-1">
          <strong>Contractor Code:</strong>
        </p>
        <p className="govuk-!-margin-0 govuk-!-margin-bottom-1">
          <strong>Active:</strong>
        </p>
        <p className="govuk-!-margin-0 govuk-!-margin-bottom-1">
          <strong>Count of SORs:</strong>
        </p>
        <p className="govuk-!-margin-0">
          <strong>Sum of SORs:</strong>
        </p>
      </div>
      <div className="govuk-!-margin-0">
        <span className="arrow right"></span>
      </div>
    </li>
    // </Link>
  )
}

export default ContractorListItem
