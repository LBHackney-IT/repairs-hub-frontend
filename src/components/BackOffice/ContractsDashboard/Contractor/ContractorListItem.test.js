import { render } from '@testing-library/react'

import ContractorListItem from './ContractorListItem'

describe('Contractors list item component', () => {
  it('should render the component', async () => {
    const { asFragment } = render(
      <ContractorListItem
        contractorReference={'SYC'}
        contractorName={'Sycous Limited'}
      />
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
