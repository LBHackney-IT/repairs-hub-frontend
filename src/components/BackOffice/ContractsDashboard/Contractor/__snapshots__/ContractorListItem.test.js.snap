// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contractors list item component should render the component 1`] = `
<DocumentFragment>
  <li
    class="govuk-!-margin-top-3 operative-work-order-list-item"
    style="cursor: pointer; border: 5px solid #00664f; border-radius: 20px; width: 85%;"
  >
    <div
      class="contract-details"
    >
      <h3
        class="lbh-heading-h3 lbh-!-font-weight-bold govuk-!-margin-bottom-1"
      >
        Sycous Limited
      </h3>
    </div>
    <div
      class="govuk-!-margin-0"
    >
      <span
        class="arrow right"
      />
    </div>
  </li>
</DocumentFragment>
`;
